traits/ctraits.c

```diff
@@ -4614,9 +4618,9 @@ get_trait_post_setattr(trait_object *trait, void *closure)
 static int
 set_trait_post_setattr(trait_object *trait, PyObject *value, void *closure)
 {
-    if (!PyCallable_Check(value)) {
+    if (value != Py_None && !PyCallable_Check(value)) {
```

Comment:
As in the comment above, if the `value` is `None` I think we want to set the `py_post_setattr` field to `NULL` instead of `Py_NONE`.